bundles/binding/org.openhab.binding.followmemusic/src/main/java/org/followmemusic/location/DatabaseLocationManager.java

```diff
@@ -0,0 +1,269 @@
+			logger.warn("New frame for id "+sensorId);
+			
+			Dimension screenSize = new Dimension(640, 480);
+						
+	        final JFrame frame = new JFrame();
+	        frame.setSize(w, h);
+	        frame.setTitle("Sensor "+sensorId);
+	        frame.setLocation(framePositionX*w, framePositionY*h);
+	        
+	        framePositionX++;
+	        if(framePositionX*w >= screenSize.getWidth()) {
+	        	framePositionX = 0;
+	        	framePositionY++;
+	        }
+	        
+	        SwingUtilities.invokeLater(new Runnable() {
+			    public void run() {
+			      // Here, we can safely update the GUI
+			      // because we'll be called from the
+			      // event dispatch thread
+			    frame.setVisible(true);
+			    }
+			});
+	        	        
+	        frames.put(sensorId, frame);
+		}
+		
+		logger.warn("Update frame for sensor "+sensorId);
+		final SignalPlot plot = new SignalPlot(signal, 0, 256);
+		
+		final JFrame frame = frames.get(sensorId);
+		
+		SwingUtilities.invokeLater(new Runnable() {
+		    public void run() {
+		      // Here, we can safely update the GUI
+		      // because we'll be called from the
+		      // event dispatch thread
+		    	frame.getContentPane().removeAll();
+				frame.getContentPane().add(plot);
+				frame.getContentPane().revalidate();
+				frame.repaint();
+		    }
+		});
+	}
+	
+	@Override
+	protected Room computeUserLocation(User user) {
+		
+		logger.info("Computing user location ...");
+		
+		LocationRepository repo = new LocationRepository(this.connection);
+		
+		// Get recent user locations
+		ArrayList<Location> locations = null;
+		try {
+			locations = repo.findRecentLocationsOfUser(user, false);
+		} catch (IncompleteResultException e) {
+			e.printStackTrace();
+		} catch (ObjectNotFoundException e) {
+			e.printStackTrace();
+		}
+		
+		if(locations != null && locations.size() > 0)
+		{
+			// Sort them by sensor
+			ArrayList<SortedSensorLocations> sortedLocations = SortedSensorLocations.sortLocations(locations);
+						
+			// Find sensors that have values higher than the threshold
+			ArrayList<Hashtable<String, Object>> thresholdSensors = new ArrayList<Hashtable<String, Object>>();
+			for(SortedSensorLocations l : sortedLocations)
+			{
+				// Get filtered values
+				float values[] = l.getOutputValues();
+				float max = 0;
+				int index = 0, i, size = values.length;
+												
+				// Study values
+				for(i=0; i<size; i++) {
+					float v = values[i];
+					
+					//Find max 
+					if(v > max) {
+						max = v;
+						//rssi id
+						index = i;
+					}
+				}
+				
+				// If there is a max > threshold
+				if(max > THRESHOLD_SENSOR_VALUE) {
+					// Save result
+					Hashtable<String, Object> val = new Hashtable<String, Object>();
+					val.put("sensor", l);
+					val.put("index", index);
+					val.put("max", max);
+					
+					thresholdSensors.add(val);
+				}
+			}
+			
+			if(thresholdSensors.size() > 0)
+			{
+				logger.warn("------------- ALGO COMPUTING -----------");
```

Comment:
at least decrease to DEBUG or better remove or enhance with reasonable debug information